### PR TITLE
Make test-kitchen with Vagrant work

### PR DESCRIPTION
**Pull request by miah**
v1 of our TK rework. This starts getting tests running on Vagrant for local testing.

### CODE REVIEW

File: kitchen.vagrant.yml [hunk 1]
```diff
@@ -14,46 +14,43 @@ lifecycle:
     - local: gem build inspec-core.gemspec --output test/kitchen/cookbooks/os_prepare/files/inspec-core-local.gem
 
 platforms:
-  - name: amazonlinux
   - name: amazonlinux-2
+    driver:
+      box: bento/amazonlinux-2
   - name: centos-6
+    driver:
+      box: bento/centos-6
```
Comment (tas50) [inline kitchen.vagrant.yml:23]: This all comes for free with kitchen-vagrant

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]
This is awesome! However, looks like we need to build `inspec-core` and then install it as such...

```
       [2020-02-10T15:00:16+00:00] FATAL: Chef::Exceptions::FileNotFound: cookbook_file[/root/inspec.gem] (run-first::default line 1) had an error: Chef::Exceptions::FileNotFound: Cookbook 'run-first' (1.0.0) does not contain a file at any of these locations:
         files/host-resources-core-ubuntu-1804.vagrantup.com/inspec.gem
         files/ubuntu-18.04/inspec.gem
         files/ubuntu/inspec.gem
         files/default/inspec.gem
         files/inspec.gem

       This cookbook _does_ contain: ['files/.gitkeep','files/inspec-core-bin.gem','files/inspec-core.gem']
``` 

Super close!

**Comment by miah** [comment]
> This is awesome! However, looks like we need to build `inspec-core` and then install it as such...
> 
> ```
>        [2020-02-10T15:00:16+00:00] FATAL: Chef::Exceptions::FileNotFound: cookbook_file[/root/inspec.gem] (run-first::default line 1) had an error: Chef::Exceptions::FileNotFound: Cookbook 'run-first' (1.0.0) does not contain a file at any of these locations:
>          files/host-resources-core-ubuntu-1804.vagrantup.com/inspec.gem
>          files/ubuntu-18.04/inspec.gem
>          files/ubuntu/inspec.gem
>          files/default/inspec.gem
>          files/inspec.gem
> 
>        This cookbook _does_ contain: ['files/.gitkeep','files/inspec-core-bin.gem','files/inspec-core.gem']
> ```
> 
> Super close!

https://github.com/inspec/inspec/pull/4895/files#diff-7390c51a77fd2549a8f2a9412e6d727aR13-R15

**Comment by miah** [comment]
> This is awesome! However, looks like we need to build `inspec-core` and then install it as such...
> 
> ```
>        [2020-02-10T15:00:16+00:00] FATAL: Chef::Exceptions::FileNotFound: cookbook_file[/root/inspec.gem] (run-first::default line 1) had an error: Chef::Exceptions::FileNotFound: Cookbook 'run-first' (1.0.0) does not contain a file at any of these locations:
>          files/host-resources-core-ubuntu-1804.vagrantup.com/inspec.gem
>          files/ubuntu-18.04/inspec.gem
>          files/ubuntu/inspec.gem
>          files/default/inspec.gem
>          files/inspec.gem
> 
>        This cookbook _does_ contain: ['files/.gitkeep','files/inspec-core-bin.gem','files/inspec-core.gem']
> ```
> 
> Super close!

Yes. I am currently having issues with `chef_gem`  or `execute` installing these together though.

**Review by zenspider — APPROVED** [review]

**Comment by james-stocks** [comment]
I'd rather not merge this and have it live in Expeditor until we have a current plan to tackle the #4422 epic - we can re-open this PR at that time.

**Pull request closed without merging**